### PR TITLE
net: introduce net_pkt_remove_tail() and fix CRC16 removal from PPP frames

### PR DESCRIPTION
**Pull request by mniestroj**
Introduce a helper function for being able to remove any arbitrary
length from tail of packet. This is handy in cases when removing
unneeded data, like CRC once it was verified.

CRC16 was removed by simply decreasing length of the last fragment by 2.
This worked as long as last fragment was longer than 1 byte. If not,
then last fragment was corrupted (its length ended up being 65535),
leading to undefined behavior.

Fix CRC16 removal by utilizing recently introduced
net_pkt_remove_tail(), that properly handles multiple fragments.

This is alternative solution to #33255, that came up during review.

### CODE REVIEW

File: include/net/net_pkt.h [hunk 1]
```diff
@@ -1705,6 +1705,19 @@ size_t net_pkt_available_payload_buffer(struct net_pkt *pkt,
...
+ * @brief Remove @a length bytes from tail of packet
+ *
+ * @details This function does not take packet cursor into account. It is a
+ *          helper to remove unneeded bytes from tail of packet (like appended
+ *          CRC). It takes care of buffer deallocation if removed bytes span
+ *          whole buffer(s).
+ *
+ * @param pkt    Network packet
+ * @param length Number of bytes to be removed
+ */
```
Comment (chrta) [inline include/net/net_pkt.h:1718]: The return value should be documented.

**Review by chrta — CHANGES_REQUESTED** [review]
I think having this function is a good addition, just one minor nit.

File: include/net/net_pkt.h (hunk 1 above)
Comment (mniestroj) [reply include/net/net_pkt.h:1721]: Good catch, thank! Documented now.

Comment (mniestroj) [reply include/net/net_pkt.h:1721]: While at it, I also added verification of return value in unit tests.

**Review by chrta — APPROVED** [review]
lgtm

**Review by rlubos — APPROVED** [review]
LGTM

**Review by cfriedt — APPROVED** [review]
LGTM. Thanks for adding to the testsuite

**Review by jukkar — APPROVED** [review]

**Pull request merged**